### ISSUE TITLE
Responsive.Showcase Connection

### ISSUE THREAD

**Issue by MichiReich** [OFFLINE-GmbH/oc-site-search-plugin#13]
I wrote a connection for the Showcase Plugin from Responsiv. Now its not working anymore. Any ideas?

```
<?php
namespace OFFLINE\SiteSearch\Classes\Providers;

use Responsiv\Showcase\Models\Item;
use Illuminate\Database\Eloquent\Collection;
use OFFLINE\SiteSearch\Models\Settings;
use System\Classes\PluginManager;

/**
 * Searches the contents generated by the
 * Responsiv.Showcase plugin
 *
 * @package OFFLINE\SiteSearch\Classes\Providers
 */
class ResponsivResultsProvider extends ResultsProvider
{
    /**
     * Runs the search for this provider.
     *
     * @return ResultsProvider
     */
    public function search()
    {
        if ( ! $this->isInstalledAndEnabled()) {
            return $this;
        }

        foreach ($this->items() as $item) {
            // Make this result more relevant, if the query is found in the title
            $relevance = stripos($item->title, $this->query) === false ? 1 : 2;

            $this->addResult($item->title, $item->description, $this->getUrl($item), $relevance);
        }

        return $this;
    }

    /**
     * Get all posts with matching title or content.
     *
     * @return Collection
     */
    protected function items()
    {
        return Item::where('title', 'like', "%{$this->query}%")
                   ->orWhere('description', 'like', "%{$this->query}%")
                   ->orWhere('content', 'like', "%{$this->query}%")
                   ->get();
    }

    /**
     * Checks if the Plugin is installed and
     * enabled in the config.
     *
     * @return bool
     */
    protected function isInstalledAndEnabled()
    {
        return true;
    }

    /**
     * Genreates the url to a blog post.
     *
     * @param $post
     *
     * @return string
     */
    protected function getUrl($post)
    {
        $url = trim(Settings::get('arrizalamin_portfolio_url', '/portfolio/project'), '/');

        return implode('/', [$url, $post->slug]);
    }

    /**
     * Display name for this provider.
     *
     * @return mixed
     */
    public function displayName()
    {
        return Settings::get('arrizalamin_portfolio_label', 'Portfolio');
    }
}
```

**Comment by tobias-kuendig** [comment]
Do you have any error messages? What doesn't work?

**Comment by tobias-kuendig** [comment]
Try adding the result like this:

```php
foreach ($this->items() as $item) {
    // Make this result more relevant, if the query is found in the title
    $relevance = stripos($item->title, $this->query) === false ? 1 : 2;

    $result        = new Result($this->query, $relevance);
    $result->title = $item->title;
    $result->text  = $item->description;
    $result->url   = $this->getUrl($item);
    $this->addResult($result);
}
```

We refactored this class a while ago. Maybe this is your problem. 

See https://github.com/OFFLINE-GmbH/oc-site-search-plugin/blob/master/classes/providers/ArrizalaminPortfolioResultsProvider.php for an example.

**Comment by MichiReich** [comment]
The Search doesn't work with the plugin Showcase. Sadly no error message.

I tried this and also to copy the ArrizalaminPorfolioResultsProvider.php and change it to the other Plugins settings. Not working either.

**Comment by tobias-kuendig** [comment]
Nothing in your `storage/logs/system.log` ?

**Comment by tobias-kuendig** [comment]
I've added an experimental search provider for Responsiv.Showcase.

I don't have a copy of Responsiv.Showcase myself so I cannot test if it actually works. Could you download [this version of SiteSearch](https://github.com/OFFLINE-GmbH/oc-site-search-plugin/archive/feature/responsiv-showcase.zip), place it into your installation and see if the provider works as expected? You can simply move your existing `plugins/offline/sitesearch` folder and put the new version in place.

Your help is greatly appreciated!

**Comment by MichiReich** [comment]
Works perfectly. Thank you so much for your support!
You can test it here: [https://mutant-library.com](Mutant Library)

**Comment by tobias-kuendig** [comment]
Released as version 1.2.6. Thanks for testing it!
